app/controllers/admins/sessions_controller.rb

```diff
@@ -1,15 +1,25 @@
 class Admins::SessionsController < Devise::SessionsController
- before_filter :check_subdomain?, only: [:new]
+  before_action :check_subdomain?, only: [:new]
+  before_action :check_admin_company_and_status, only: [:create]
 
   protected
 
   def check_subdomain?
     subdomain = request.subdomain
-    if ['www', 'ftp', 'ssh'].include?(subdomain)
-      return redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
+    if subdomain == 'www'
+      redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
```

Comment:
if we don't pass host then it'll keep on checking the subdomain again and again on every request so thats why host is used directly so that it doesn't check subdomain to the redirected request.